apps/nova/src/components/Layout/Sidebar.tsx

```diff
@@ -128,15 +136,29 @@ const Sidebar = () => {
           {intl("sidebar.main")}
         </Typography>
         <ul className="flex flex-col gap-2">
-          {mainPages.map(({ label, route, Icon }) => (
-            <SidebarItem
-              key={label}
-              to={route}
-              Icon={Icon}
-              label={label}
-              active={location.pathname === route}
-            />
-          ))}
+          {mainPages.map(({ label, route, Icon }) => {
+            if (user?.role === IUser.Role.Tutor && route === Route.Subscription)
+              return;
+
+            if (user?.role === IUser.Role.Tutor && route === Route.Tutors)
+              return;
+
+            if (
+              user?.role === IUser.Role.Student &&
+              route === Route.ScheduleManagement
+            )
+              return;
```

Comment:
This is not the right place to add these checks. The checks should be in the previous step when constructing list in the first place. 

The logic should go to `useMemo` for `mainPages` . 

Inside the `useMemo` their should be something like this.

```
const pages = [{...}];

if (tutor || tutorManager) pages.push({...}); 
if (student) pages.push({...})
```